core/json/src/main/scala/net/liftweb/json/Extraction.scala

```diff
@@ -319,14 +331,54 @@ object Extraction {
       }
     }
 
+    def newTuple(root: JValue, mappings: List[Mapping]): Any = {
+      root match {
+        case JArray(items) if items.length >= 1 && items.length <= 22 =>
+          val builtItems: Seq[Object] = items.zip(mappings).map({
+            case (i,m) =>
+              build(i, m).asInstanceOf[Object]
+          })
+          val tupleClass = Class.forName("scala.Tuple" + items.length)
+
+          val typedTupleConstructor = tupleClass.getConstructors()(0)
+          typedTupleConstructor.newInstance(builtItems: _*)
+
+        case JObject(items) if items.forall(_.name.startsWith("_")) =>
+          newTuple(JArray(items.map(_.value)), mappings)
```

Comment:
We probably need to sort these. For example, if we get a `{"_3": 5, "_1": "hello", "_2": "uh-oh"}`, that should deserialize to `Tuple3[String,String,Int]("hello", "uh-oh", 5)`, but I think with this code we'd get `Tuple3[Int,String,String](5, "hello", "uh-oh")`.